src/main/java/seedu/address/logic/Logic.java

```diff
@@ -51,7 +53,17 @@ public interface Logic {
     void setGuiSettings(GuiSettings guiSettings);
 
     /**
-     * Returns the user input history.
+     * Returns the EventBook.
+     *
+     * @see seedu.address.model.Model#getEventBook()
+     */
+    ReadOnlyEventBook getEventBook();
+
+    /** Returns an unmodifiable view of the filtered list of persons */
+    ObservableList<Event> getFilteredEventList();
+
+    /**
+     * Returns the user prefs' address book file path.
      */
-    UserInputHistory<String> getUserInputHistory();
```

Comment:
Why do you remove other's code? This is a feature for traversing through old commands.